lib/widgets/resources/motor.dart

```diff
@@ -42,20 +42,17 @@ class _ViamMotorWidgetState extends State<ViamMotorWidget> {
       });
       await Future.delayed(const Duration(milliseconds: 10));
       if (await widget.motor.isMoving()) {
-        if (power == 0) {
```

Comment:
Trying to understand what's going on here. So the previous behavior of `stop` was that we'd try stopping the motor (and setting power state on the widget to 0). Then, we'd wait 10ms. Then, we'd see if the motor was still moving, and only attempt a `stop` again if the power state was still 0? Assuming the user could have altered the power state within those 10ms and no longer want the motor to stop?

Why do we need to wait 10ms at all? Why can't the definition of this method just be:
```
Future<void> stop() async {
  try {
    await widget.motor.stop();
    setState(() {
      power = 0;
    });
  } catch (e) {
    error = e as Error;
  }
}
```

Reply:
If I remember correctly the the 10ms pause and then checking again was an attempt to solve this same bug when it was present in the example app. Back then it wasn't honoring the stop call sometimes when the user released their touch, so this was an attempt to ensure it actually stopped.

Reply:
Gotcha. I guess I wonder why the first stop doesn't cut it, but the workaround makes sense.

Reply:
My theory is because of the large amount of setPower calls we're making. maybe somehow that overloads things and makes a race condition. We could limit those calls and look for a certain percentage of change or allow so many calls per second if thats the case, which might be a better solution. But I haven't looked into it enough to confirm.

Reply:
I wish I knew. My tests show that the system just doesn't set the power to zero, or that there's some weird touch behavior where the power is set to zero, but the UI pushes it back to whatever it used to be and negates the release.